libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp

```diff
@@ -358,25 +367,19 @@ String ESP8266WiFiAPClass::softAPmacAddress(void) {
 String ESP8266WiFiAPClass::softAPSSID() const {
     struct softap_config config;
     wifi_softap_get_config(&config);
-    char* name = reinterpret_cast<char*>(config.ssid);
-    char ssid[sizeof(config.ssid) + 1];
-    memcpy(ssid, name, sizeof(config.ssid));
-    ssid[sizeof(config.ssid)] = '\0';
-
-    return String(ssid);
+    String ssid;
+    ssid.concat(reinterpret_cast<const char*>(config.ssid), config.ssid_len);

```

Comment:
Just to note, with this test API still returns non-zero length field when user sets ssid_len=0:
```diff
diff --git a/libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp b/libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp
index 392e142e..9fa2e514 100644
--- a/libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp
+++ b/libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp
@@ -123,7 +123,9 @@ bool ESP8266WiFiAPClass::softAP(const char* ssid, const char* passphrase, int ch
     if (ssid_len < 32) {
         conf.ssid[ssid_len] = '\0';
     }
-    conf.ssid_len = ssid_len;
+    conf.ssid_len = 0;
+    Serial.printf("conf.ssid_len=%hhu\n", conf.ssid_len);
+    //conf.ssid_len = ssid_len;

     if(!passphrase || pass_len == 0) {
         conf.authmode = AUTH_OPEN;
```
```cpp
#include <Arduino.h>
void setup() {
    Serial.begin(115200);
    auto timeout = esp8266::polledTimeout::oneShotFastMs(1000);
    while (!timeout) {
        delay(100);
    }

    WiFi.persistent(false);
    WiFi.mode(WIFI_AP);
    String ssid;
    while (ssid.length() < 31) {
        ssid += 'A';
    }
    String pass("testtesttest");

    WiFi.softAP(ssid, pass, 6);
    delay(100);
    
    softap_config config;
    wifi_softap_get_config(&config);
    Serial.printf("retrieved ssid_len=%hhu\n", config.ssid_len);
}
void loop() {
}
```
```
conf.ssid_len=0
retrieved ssid_len=31
```